### ISSUE TITLE
TELNET thinks ES is a CHAOSNET-only host

### ISSUE THREAD

**Issue by eswenson1** [PDP-10/its#1552]
If I attempt to open a connection to the web serser running on ES using TELNET, and specify ES by hostname, TELNET refuses, claiming that ES is a chaosnet-only host:
```
telnet↑K!
User TELNET.752
TELNET>Open connection to 120,ES
 This program cannot reach that Chaosnet-only host.
TELNET>
```
However, if I specify the IP address of ES (as seen by ES), it works:
```
TELNET>Open connection to 120,192.168.0.100
 Trying... Open
GET /
HTTP/1.0 200 ask and thou shalt receive
Content-Type: text/html

<html><body><h1>ITS works!</h1></body></html>
βββ
Connection closed;
TELNET>
```
Why does TELNET think ES is CHAOSNET only? The host table has this entry:
```
;;; ES-ITS
HOST : CHAOS 5460, 192.168.0.100 : ES-ITS.SWENSON.ORG, ES : PDP-10 : ITS : :
```

**Comment by eswenson1** [comment]
When TELNET invokes HOSTNM (in SYSNET;NETWRK >) to read in a host name, that routine returns the host address in A. The network number is in the high bits of this address.  If the host is on multiple networks (chaosnet and internet), as ES is, HOSTNM "prefers" chaosnet and returns this as the network number.   However, TELNET treats this value as indicating that the host is *ONLY* on this network. In my case, ES is on two networks.  This TELNET logic is flawed.

**Comment by bictorv** [comment]
Any IP address is on the "arpanet" now, isn't it?

**Comment by eswenson1** [comment]
Well, the code checks the host table for the definition of each network based on either 1, 2, or 3 octet a of the up address. The arpanet is, according to the network routines, network 10. If you’re ITS network address is on network 10, it is considered to be on the arpanet. Otherwise not. We should fix this code to assume a host is on the arpanet if its address is not a chaosnet address.

**Comment by bictorv** [comment]
Yes, that's what I meant: all hosts with an IP address should be assumed to be "arpa". But be careful about the IP net which means "this is a Chaosnet address", I forgot which.

**Comment by larsbrinkhoff** [comment]
I vaguely and maybe wrongly recall the network addresses has a bit that
say they're "not internet", or "unternet".

**Comment by larsbrinkhoff** [comment]
This is in H3TEXT:

    NET : UN 7.0.0.0 : CHAOS :

It shouldn't be possible to confuse this with IP address 7.0.0.0 because
of the UN bit.

**Comment by eswenson1** [comment]
Right. I think is 7 or 6.

**Comment by eswenson1** [comment]
Ah. UN + 7. Should be an easy fix.
